tests/host_tools/fcmetrics.py

```diff
@@ -363,3 +365,135 @@ def validate_per_device_metrics(self, fc_metrics):
+        if key.endswith("_us"):
+            return "Microseconds"
+    return "Count"
+
+
+def flush_fc_metrics_to_cw(fc_metrics, metrics):
+    """
+    Flush Firecracker metrics to CloudWatch
+    Use a new metrics logger and existing dimensions
+    Add a prefix "fc_metrics." to differentiate these metrics.
+    NOTE:
+        There are metrics with keywords "fail", "err",
+        "num_faults", "panic" in their name and represent
+        some kind of failure in Firecracker.
+        This function `does not` assert on these failure metrics
+        since some tests might not want to assert on them while
+        some tests might want to assert on some but not others.
+    """
+    # List of SharedStoreMetric that once updated have the same value thoughout the life of vm
+    metrics_to_export_once = {
+        "api_server.process_startup_time_us",
+        "api_server.process_startup_time_cpu_us",
+        "latencies_us.full_create_snapshot",
+        "latencies_us.diff_create_snapshot",
+        "latencies_us.load_snapshot",
+        "latencies_us.pause_vm",
+        "latencies_us.resume_vm",
+        "latencies_us.vmm_full_create_snapshot",
+        "latencies_us.vmm_diff_create_snapshot",
+        "latencies_us.vmm_load_snapshot",
+        "latencies_us.vmm_pause_vm",
+        "latencies_us.vmm_resume_vm",
+    }
+    skip = set()
+    for group, keys in fc_metrics.items():
+        if group == "utc_timestamp_ms":
+            continue
+        for key, value in keys.items():
+            full_key = f"{group}.{key}"
+            # values are 0 when:
+            # - there is no update
+            # - device is not used
+            # - SharedIncMetric reset to 0 on flush so if
+            #   there is no change metric the values remain 0.
+            # We can save the amount of bytes we export to
+            # CloudWatch in these cases.
+            # however it is difficult to differentiate if a 0
+            # should be skipped or upload because it could be
+            # an expected value in some cases so we upload
+            # all the metrics even if data is 0.
+            if full_key in skip:
+                continue
+            unit = get_emf_unit_for_fc_metrics(key)
+            metrics.put_metric(f"fc_metrics.{full_key}", value, unit=unit)
+            # print(f"fc_metrics.{full_key} {value}, unit={unit}")
+            if full_key in metrics_to_export_once:
+                skip.add(full_key)
+
+
+class FCMetricsMonitor(Thread):
+    """
+    read Firecracker metrics from the microvm every 60secs and
+    uploads the metrics to CW.
+    We do this as a daemon thread every 60sec, instead of collecting
+    all metrics together in the end, to retain timestamp of the metrics.
+    """
+
+    def __init__(self, vm, metrics_logger, timer=60):
+        Thread.__init__(self, daemon=True)
+        self.vm = vm
+        self.timer = timer
+
+        self.metrics_index = 0
+        self.running = True
```

Comment:
nit: hmm should this be False here and then set to True as the 1st line in `run`?